src/storage/DataAccessorBasedStore.ts

```diff
@@ -0,0 +1,314 @@
+import type { Readable } from 'stream';
+import { DataFactory } from 'n3';
+import type { Quad } from 'rdf-js';
+import streamifyArray from 'streamify-array';
+import { v4 as uuid } from 'uuid';
+import type { Patch } from '../ldp/http/Patch';
+import type { Representation } from '../ldp/representation/Representation';
+import { RepresentationMetadata } from '../ldp/representation/RepresentationMetadata';
+import type { ResourceIdentifier } from '../ldp/representation/ResourceIdentifier';
+import { INTERNAL_QUADS } from '../util/ContentTypes';
+import { ConflictHttpError } from '../util/errors/ConflictHttpError';
+import { MethodNotAllowedHttpError } from '../util/errors/MethodNotAllowedHttpError';
+import { NotFoundHttpError } from '../util/errors/NotFoundHttpError';
+import { UnsupportedHttpError } from '../util/errors/UnsupportedHttpError';
+import type { MetadataController } from '../util/MetadataController';
+import { CONTENT_TYPE, HTTP, LDP, RDF } from '../util/UriConstants';
+import { ensureTrailingSlash, trimTrailingSlashes } from '../util/Util';
+import type { DataAccessor } from './accessors/DataAccessor';
+import type { ContainerManager } from './ContainerManager';
+import type { ResourceStore } from './ResourceStore';
+
+/**
+ * ResourceStore which uses a DataAccessor for backend access.
+ *
+ * This class is important because this way different stores don't all need to implement behaviour
+ * Examples:
+ *  * Converting container metadata to data
+ *  * Converting slug to URI
+ *  * Checking if addResource target is a container
+ *  * Checking if no containment triples are written to a container
+ *  * etc.
+ *
+ * Currently "metadata" is seen as something that is not directly accessible.
+ * That means that a user can't write directly to the metadata of a resource, only indirectly through headers.
+ * (Except for containers where data and metadata overlap).
+ *
+ * The one thing this store does not take care of (yet?) are containment triples for containers
+ *
+ * Work has been done to minimize the amount of required calls to the DataAccessor,
+ * but the main disadvantage is that sometimes multiple calls are required where a specific store might only need one.
+ */
+export class DataAccessorBasedStore implements ResourceStore {
```

Comment:
Hard to follow the logic of the main methods of this class. Missing some comments and/or clear block delineations?